### PR TITLE
PD-545: Add new types such that usermenu can be used without mongonav

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes
- University needs a Menu to use ASAP, and we've deprecated MongoMenu, so we need to make UserMenu resilient when used in environments outside of the MongoNav component
- Only blocker here is that the Hosts/URLs are currently required for the entire MongoNav component, so suggested changes.. 

- Hosts/URLs are no longer required props
- Duplicated logic for creating defaults for Hosts and URLs from MongoNav, as if this component is to be used outside of the MongoNav component, it will need this logic implemented
- To Note: I know there is a PR floating around out there made by @seanmhanson which moves some of this URL logic outside of the <MongoNav /> component, so I didn't want to duplicate this work right now and was thinking we could merge without the more generalized defaults and then update once their PR is in

🎟 _Jira ticket:_ [PD-545](https://jira.mongodb.org/browse/[PD-545])

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### CODE REVIEW

File: packages/mongo-nav/src/user-menu/UserMenu.tsx [hunk 1]
```diff
@@ -22,6 +23,15 @@ import {
...
 } from '../types';
 
+const defaultHosts: Required<HostsInterface> = {
+  account: 'https://account.mongodb.com',
+  cloud: 'https://cloud.mongodb.com',
+  university: 'https://university.mongodb.com',
+  support: 'https://support.mongodb.com',
+  charts: 'https://charts.mongodb.com',
+  realm: 'https://stitch.mongodb.com',
+};
```
Comment (DesignerDave) [inline packages/mongo-nav/src/user-menu/UserMenu.tsx]: Should this have an `as const`?

File: packages/mongo-nav/src/user-menu/UserMenu.tsx [hunk 2]
```diff
@@ -22,6 +23,15 @@ import {
   HostsInterface,
 } from '../types';
 
+const defaultHosts: Required<HostsInterface> = {
```
Comment (DesignerDave) [inline packages/mongo-nav/src/user-menu/UserMenu.tsx]: This doesn't need to be marked `Required` I don't believe.

File: packages/mongo-nav/src/user-menu/UserMenu.tsx [hunk 3]
```diff
@@ -154,19 +164,74 @@ interface UserMenuProps {
...
   onLogout = () => {},
   onProductChange = () => {},
-  urls,
-  hosts,
+  urls: urlsProp,
+  hosts: hostsProp,
 }: UserMenuProps) {
+  const hosts = defaultsDeep(hostsProp, defaultHosts);
+
+  const defaultURLs = {
```
Comment (DesignerDave) [inline packages/mongo-nav/src/user-menu/UserMenu.tsx]: Might wanna drop a comment that this is temporary.

File: packages/mongo-nav/src/user-menu/UserMenu.tsx (hunk 1 above)
Comment (brookescarlett) [reply packages/mongo-nav/src/user-menu/UserMenu.tsx]: 👍

File: packages/mongo-nav/src/user-menu/UserMenu.tsx (hunk 2 above)
Comment (brookescarlett) [reply packages/mongo-nav/src/user-menu/UserMenu.tsx]: this is just the defaults and we are supplying them, so we should be able to leave it as required

File: packages/mongo-nav/src/user-menu/UserMenu.tsx (hunk 3 above)
Comment (brookescarlett) [reply packages/mongo-nav/src/user-menu/UserMenu.tsx]: 👍

File: packages/mongo-nav/src/user-menu/UserMenu.tsx (hunk 2 above)
Comment (hswolff) [reply packages/mongo-nav/src/user-menu/UserMenu.tsx]: Chatted with Brooke offline- this is temporary and another PR unifies these data structures that are used in multiple places. So this works for me for now, so long as we clean it up later.

Comment (DesignerDave) [reply packages/mongo-nav/src/user-menu/UserMenu.tsx]: SGTM, resolving.

**Pull request merged**